grapple/types/rich_text.py

```diff
@@ -0,0 +1,19 @@
+from django.template.loader import render_to_string
+from graphene.types import Scalar
+
+try:
+    from wagtail.rich_text import expand_db_html
+except ImportError:
+    from wagtail.core.rich_text import expand_db_html
+
+from ..settings import grapple_settings
+
+
+class RichText(Scalar):
+    @staticmethod
+    def serialize(rich_text: str):
+        if grapple_settings.RICHTEXT_FORMAT == "html":
+            return render_to_string(
+                "wagtailcore/richtext.html", context={"html": expand_db_html(rich_text)}
+            )
+        return rich_text
```

Comment:
I have not implemented `parse_literal` or `parse_value` (see https://docs.graphene-python.org/en/stable/types/scalars/#custom-scalars). If I understand correctly these are needed to make the type writable. Should they be implemented?

Reply:
Also related to this method: the `== "html"` case could be simplified to `WagtailRichText(rich_text).__html__()`, however grapple is using its own `richtext.html` template.

The grapple template was introduced in https://github.com/torchbox/wagtail-grapple/pull/90, to avoid adding `<div class="rich-text">...</div>` to the output, which was included in the Wagtail `richtext.html` template at the time.

The wrapping div was removed from Wagtail for 2.10 (see https://github.com/wagtail/wagtail/commit/a1a2c35c1c1e78efa38bed0074ae0c7968770e84). However, grapple's `richtext.html` contains a trailing newline, where Wagtail's does not.

Should we remove the template from grapple, and use wagtail's? If so, should I open a new ticket for it or make the change in this PR?

Reply:
I would be all up for this, however `wagtail.contrib.legacy.richtext` gives me pause.
saying that, it should not be an issue on headless sites, so let's do the requisite tidy up and add a good nice notice for upgrades